scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1947,65 +1961,80 @@ private void sendMessage(final long nanos, final RawData message, final Connecti
 				initializeHandshaker(handshaker);
 				handshaker.startHandshake();
 			}
+			message.onConnecting();
 			handshaker.addApplicationDataForDeferredProcessing(message);
-		}
-		else {
-			Long timeout = autoResumptionTimeoutMillis;
-			String contextTimeout = message.getEndpointContext().get(DtlsEndpointContext.KEY_RESUMPTION_TIMEOUT);
-			if (contextTimeout != null) {
-				if (contextTimeout.isEmpty()) {
-					timeout = null;
-				} else {
-					try {
-						timeout = Long.valueOf(contextTimeout);
-					} catch (NumberFormatException e) {
-					}
-				}
-			}
-			if (connection.isAutoResumptionRequired(timeout)) {
-				// create the session to resume from the previous one.
-				if (serverOnly) {
-					message.onError(new EndpointUnconnectedException("server only, resumption required!"));
+		} else {
+			if (none) {
+				if (connection.isResumptionRequired()) {
+					message.onError(new EndpointUnconnectedException("resumption requested failed!"));
```

Comment:
"resumption requested failed!" sounds like if this is an error.
Maybe just "session without connection". ?

Reply:
There are many reasons for "resumption required".
May be that's also a good message: "resumption required!" ?